### PR TITLE
fix(ci): allow scanner v4 installation to be configurable

### PR DESCRIPTION
**Pull request by BradLugo**
## Description

These changes allow the operator install of scanner v4 to be configurable, defaulting to being disabled.

Related to https://github.com/openshift/release/pull/48344

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

CI

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by mclasmeier** [comment]
Can we keep the support built in and just not run it?

**Comment by BradLugo** [comment]
Sorry, I'm not following. Can you rephrase the question?

**Review by dcaravel — COMMENTED** [review]
Changing the `ROX_SCANNER_V4` env var tells `central` (and `sensor`) to behave as if `Scanner V4` does not exist, however the `scanner-v4-*` components are still installed.

In the past 8ish hours, there have been CI failures where central cannot be scheduled (no nodes with appropriate resources), and also suspicious log entries in scanner-v4 components.  Since scanner-v4 is not currently 'under test' , it may be best to not install scanner-v4 at all.

Thoughts on instead of setting the `ROX_SCANNER_V4` `env var` to use the value to change the installer configuration used by CI to explicitly disable installing any scanner v4 components?

**Comment by BradLugo** [comment]
> Thoughts on instead of setting the ROX_SCANNER_V4 env var to use the value to change the installer configuration used by CI to explicitly disable installing any scanner v4 components?

Agreed. This is what I thought my changes did, but I see now it only changes the central environment variable. I'll see what we can do to change the CRD itself.

> Or if using deploy scripts set ROX_SCANNER_V4 = false which will set the appropriate value for helm

This should be covered in https://github.com/openshift/release/pull/48344

> As an alternative we could address the resource issue in CI and suspicious log entries separately and keep this as is.

After these changes are merged (this PR and the OSCI PR), I'm going to submit another OSCI PR to add a variant that turns on scanner v4 but doesn't run the tests unless you explicitly tell it to. This will allow us to work on getting tests with scanner v4 buttoned up without blocking others

File: tests/e2e/lib.sh [hunk 1]
```diff
@@ -146,7 +146,6 @@ export_test_environment() {
     ci_export ROX_ADMINISTRATION_EVENTS "${ROX_ADMINISTRATION_EVENTS:-true}"
     ci_export ROX_POLICY_CRITERIA_MODAL "${ROX_POLICY_CRITERIA_MODAL:-true}"
     ci_export ROX_TELEMETRY_STORAGE_KEY_V1 "DISABLED"
-    ci_export ROX_SCANNER_V4_SUPPORT "${ROX_SCANNER_V4_SUPPORT:-true}"
```
Comment (mclasmeier) [inline tests/e2e/lib.sh:149]: This is what I have been referring to. I was about to explain why I think we should keep this one. But then I realized that the corresponding feature flag `ScannerV4Support` is an **unchangeable** feature flag nowadays, hence it shouldn't matter what `ROX_SCANNER_V4_SUPPORT` is set to in the environment. (Correct, @dcaravel?)

Hence, should be fine to remove as a cleanup.

**Review by mclasmeier — APPROVED** [review]

File: tests/e2e/lib.sh (hunk 1 above)
Comment (BradLugo) [reply tests/e2e/lib.sh:149]: Oh, I see. Yeah, I couldn't find a reference to this variable in any of the shell scripts. Do you know how that variable is propagated? It looked like `ROX_SCANNER_V4` was the variable that controlled all the scanner v4 enabling.

Comment (dcaravel) [reply tests/e2e/lib.sh:149]: > This is what I have been referring to. I was about to explain why I think we should keep this one. But then I realized that the corresponding feature flag ScannerV4Support is an unchangeable feature flag nowadays, hence it shouldn't matter what ROX_SCANNER_V4_SUPPORT is set to in the environment. (Correct, @dcaravel?)

Agreed

**Review by dcaravel — APPROVED** [review]

File: tests/e2e/lib.sh (hunk 1 above)
Comment (dcaravel) [reply tests/e2e/lib.sh:149]: Suppose to further qualify - unchangeable feature flags are changeable in non-release builds - in this case there are no plans to change the default for `ROX_SCANNER_V4_SUPPORT` so OK with the removal, or if prefer to be more explicit OK keeping it as well.

Comment (BradLugo) [reply tests/e2e/lib.sh:149]: It's still not clear to me what `ROX_SCANNER_V4_SUPPORT` does so I feel compelled to keep it but default it to false. If anyone feels differently (i.e., changing it back to true or removing it entirely), please feel free to submit that change 🙂

Comment (mclasmeier) [reply tests/e2e/lib.sh:149]: I think since there are no plans to change the defaults for ROX_SCANNER_V4_SUPPORT, removal would be preferable.

I would advice against setting it to false, because in that case non-release builds and release-builds would -- IIUC -- actually test slightly different code paths in several places.

I see no good reason for setting to false. In a nutshell: `ROX_SCANNER_V4_SUPPORT` can be described as the build-time flag for Scanner V4 *support* in different components (Sensor, Central, Helm Charts, etc.). When enabled (the "semi-unchangeable" default) these components know in principle how to communicate with Scanner V4 or how to install it.

This is clearly different from *enabling* or *actively using* Scanner V4, which is controlled by `ROX_SCANNER_V4`. For example, when this is enabled, Central and Sensor expect that there is a Scanner V4 deployment and tries talking to it.

Comment (BradLugo) [reply tests/e2e/lib.sh:149]: Understood. Thank you for the detailed explanation 🙂
